sample/src/main/java/io/instaleap/statusprogressbar/MainViewModel.kt

```diff
@@ -0,0 +1,136 @@
+        dataModelStateList.add(
+            DataModelState(
+                "added",
+                (_addedQuantity.value ?: ZERO_ITEMS),
+                StateColor.Added.color
+            )
+        )
+        dataModelStateList.add(
+            DataModelState(
+                "pendidng",
+                (_pendingQuantity.value ?: ZERO_ITEMS),
+                StateColor.Pending.color
+            )
+        )
+        dataModelStateList.add(
+            DataModelState(
+                "removed",
+                (_removedQuantity.value ?: ZERO_ITEMS),
+                StateColor.Removed.color
+            )
+        )
+
+        _dataModelView.postValue(
+            DataModelView(
+                dataList = dataModelStateList,
+                totalValue = getAllItemsQuantity()
+            )
+        )
+    }
+
+    private fun getAllItemsQuantity(): Int {
+        val totalItems = (_addedQuantity.value ?: ZERO_ITEMS) + (_removedQuantity.value ?: ZERO_ITEMS) + (_pendingQuantity.value ?: ZERO_ITEMS)
+        return if(totalItems > 0) totalItems else DEFAULT_ITEMS
+    }
+
+    private fun setListDataModelView() {
+        val dataModelList = listOf(
+            DataModelView(
+                dataList = listOf(
+                    DataModelState("added", 5, StateColor.Added.color),
+                    DataModelState("pendidng", 15, StateColor.Pending.color),
+                    DataModelState("removed", 5, StateColor.Removed.color),
+                ),
+                totalValue = 20
+            ),
+            DataModelView(
+                dataList = listOf(
+                    DataModelState("added", 5, StateColor.Added.color),
+                    DataModelState("pendidng", 5, StateColor.Pending.color),
+                    DataModelState("removed", 5, StateColor.Removed.color),
+                ),
+                totalValue = 15
+            ),
+            DataModelView(
+                dataList = listOf(
+                    DataModelState("added", 0, StateColor.Added.color),
+                    DataModelState("pendidng", 28, StateColor.Pending.color),
+                    DataModelState("removed", 2, StateColor.Removed.color),
+                ),
+                totalValue = 30
+            ),
+            DataModelView(
+                dataList = listOf(
+                    DataModelState("added", 2, StateColor.Added.color),
+                    DataModelState("pendidng", 6, StateColor.Pending.color),
+                    DataModelState("removed", 2, StateColor.Removed.color),
+                ),
+                totalValue = 10
+            ),
+            DataModelView(
+                dataList = listOf(
+                    DataModelState("added", 0, StateColor.Added.color),
+                    DataModelState("pendidng", 3, StateColor.Pending.color),
+                    DataModelState("removed", 0, StateColor.Removed.color),
+                ),
+                totalValue = 3
+            )
```

Comment:
Please use constant values

Reply:
Here could be any value, this doesn´t affect the behavior of Library, so I don't know if is necesary to set as constants. Going to set a function to get the totalValue.